File: dataset_reduzido/codenet/Java/Code/s933313035.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int m = sc.nextInt();

        Node tree = null;
        for (int i = 0; i < m; i++) {
            String command = sc.next();

            if (command.equals("insert")) {
                long key = sc.nextLong();
                if (tree == null) {
                    tree = new Node();
                    tree.value = key;
                } else {
                    tree.insert(key);
                }
            } else {
                if (tree != null) {
                    tree.printInOrder();
                    System.out.println();
                    tree.printPreOrder();
                    System.out.println();
                }
            }
        }
    }

    static class Node {
        long value;
        Node left;
        Node right;

        void insert(long newValue) {
            if (newValue < this.value) {
                if (left == null) {
                    left = new Node();
                    left.value = newValue;
                } else {
                    left.insert(newValue);
                }
            } else {
                if (right == null) {
                    right = new Node();
                    right.value = newValue;
                } else {
                    right.insert(newValue);
                }
            }
        }

        void printInOrder() {
            if (left != null) {
                left.printInOrder();
            }
            System.out.print(" " + value);
            if (right != null) {
                right.printInOrder();
            }
        }

        void printPreOrder() {
            System.out.print(" " + value);
            if (left != null) {
                left.printPreOrder();
            }
            if (right != null) {
                right.printPreOrder();
            }
        }
    }
}

File: output/Llama-2-100/codenet/Java/Python/s933313035.py
import sys

class Node:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

    def insert(self, new_value):
        if new_value < self.value:
            if self.left is None:
                self.left = Node(new_value)
            else:
                self.left.insert(new_value)
        else:
            if self.right is None:
                self.right = Node(new_value)
            else:
                self.right.insert(new_value)

    def printInOrder(self):
        if self.left is not None:
            self.left.printInOrder()
        print(" " + str(self.value))
        if self.right is not None:
            self.right.printInOrder()

    def printPreOrder(self):
        print(" " + str(self.value))
        if self.left is not None:
            self.left.printPreOrder()
        if self.right is not None:
            self.right.printPreOrder()

def main():
    n = int(input())
    scanner = Scanner(sys.stdin)

    tree = Node(0)

    for i in range(n):
        command = scanner.next()

        if command == "insert":
            key = scanner.nextLong()
            tree.insert(key)

        elif command == "printInOrder":
            tree.printInOrder()

        elif command == "printPreOrder":
            tree.printPreOrder()

if __name__ == "__main__":
    main()


